streams/src/main/java/org/apache/kafka/streams/kstream/internals/KStreamSlidingWindowAggregate.java

```diff
@@ -50,17 +59,27 @@
     private final SlidingWindows windows;
     private final Initializer<VAgg> initializer;
     private final Aggregator<? super KIn, ? super VIn, VAgg> aggregator;
+    private final EmitStrategy emitStrategy;
 
     private boolean sendOldValues = false;
 
     public KStreamSlidingWindowAggregate(final SlidingWindows windows,
                                          final String storeName,
                                          final Initializer<VAgg> initializer,
                                          final Aggregator<? super KIn, ? super VIn, VAgg> aggregator) {
+        this(windows, storeName, EmitStrategy.onWindowUpdate(), initializer, aggregator);
```

Comment:
Would it be cleaner to remove this constructor and force the caller to alway pass in an `EmitStrategy`?

Reply:
This function is now used by the co-group API which does not add emit strategy yet, so I think we should still keep it for now.